### PR TITLE
Fix credential creation from `jaws new project` prompts

### LINKED ISSUE #152: Creation of aws credentials files does not function
When running jaws on a fresh system without the admin credentials setup yet you get the following error while after filling out all of the information for `jaws new project`

```
Error: ENOENT: no such file or directory, open '/Users/boushley/.aws/credentials'
    at Error (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.readFileSync (fs.js:397:15)
    at Object.readFileSync (/Users/boushley/projects/jaws/node_modules/aws-sdk/lib/util.js:102:35)
    at Object.module.exports.profilesMap (/Users/boushley/projects/jaws/lib/utils/aws.js:73:38)
    at /Users/boushley/projects/jaws/lib/commands/new_project.js:231:37
    at tryCatcher (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/promise.js:581:18)
    at Async._drainQueue (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/boushley/projects/jaws/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:374:17)
```

I've got a branch that works past that [on my fork](https://github.com/boushley/JAWS/tree/fix-credential-creation) however it then runs into a problem where the `.aws/credentials` file gets created but is empty.

### PR DESCRIPTION
**Pull request by boushley**
This should resolve #152

### CODE REVIEW

**Comment by boushley** [comment]
The commit messages pretty well describe the problem and solution.

**Comment by boushley** [comment]
Also note that I added the [debug](https://www.npmjs.com/package/debug) package as a part of this.

I found it super helpful, and it shouldn't affect people as they're using the project.

**Comment by doapp-ryanp** [comment]
Great catch and I like the debug module. One request for you:

Would you be willing to rework the [`logIfVerbose`](https://github.com/jaws-framework/JAWS/blob/master/lib/utils/index.js#L185) command to leverage the `debug` module? This way we don't have to require the debug module all over the code (most files already require the jaws util) and it gives us the ability in the future to change how we log verbose/debug level in one spot.

**Comment by boushley** [comment]
I could change `logIfVerbose` to use `debug` however we might lose some of the functionality of `debug` unless we change `logIfVerbose` to also take a "scope" to which things are being printed.

I was figuring we could use a standard for requiring debug where you do `require('debug')('jaws:file_description')` where `file_description` is a short name of the file. This allows you to debug only certain parts of the output.

You can run: `DEBUG=jaws* jaws foo` if you want to debug everything, but you could run `DEBUG=jaws:util* jaws foo` if you only want to see debug statements from the utils.

Definitely open to patching `logIfVerbose` so that it works with `debug`, however it might be nicer to actually require debug in all the files as we need it.

Thoughts?

**Comment by doapp-ryanp** [comment]
yea I am totally cool with adding scope to `logIfVerbose` - I honestly threw it in as a quick hack.  

Here is what I'm thinking - rename+refactor `logIfVerbose` to `jawsDebug`. `jawsDebug` takes scope as the 1st param, then a var list of params (just like debug does).

Then inside `jawsDebug` you can:

```
var debug = require('debug')('jaws:'+scope);  //always prepend jaws
```

This way you can just do the following in any jaws code:

```
...
utils.jawsDebug('deploy_lambda','hi there',somObj);
...
```

You will also then need to modify [this line](https://github.com/jaws-framework/JAWS/blob/master/bin/jaws#L305) to set the DEBUG env var

Make sense? or am I missing something?

**Comment by boushley** [comment]
I'll do some testing to make sure that debug doesn't mind creating debug loggers multiple times, but that seems reasonable.

**Comment by doapp-ryanp** [comment]
Awesome dude, thank you SO much!

**Comment by ac360** [comment]
:+1: :+1: :+1:

**Comment by boushley** [comment]
Alright, I updated `logIfVerbose` to be `jawsDebug`.

Take a look and let me know what you think.

In files where we log a lot it seems a bit verbose to repeat the context over and over. Might want to create a factory function that creates a function you call without the context similar to the way debug works... but then we're very similar to just requiring debug directly. Thoughts?

**Comment by doapp-ryanp** [comment]
Nice work.

So how about this: we makde `jawsDebug` NOT take a context and instead infer the caller name using [this method](http://stackoverflow.com/questions/13227489/how-can-one-get-the-file-path-of-the-caller-function-in-node-js).  I propose we do not include the '.js' extension of the calling method file name.

We then add a new method called `utils.jawsDebugContext('somethingCool')` that takes a context.  This way 99% of our calls we dont have to specify what file we are in, and for special cases that we do want to specify a different context we have a method todo so.

make sense? If you agree would you be able to make the change? `getCaller` and `getStack` can be private (non exported) funcs in utils.  The performance of these methods are obv shitty, but doesn't matter since we are in the 'verbose' mode.

**Comment by boushley** [comment]
@doapp-ryanp great idea.

I've refactored to the new syntax. I think it looks pretty clean.

**Comment by doapp-ryanp** [comment]
@boushley nice work. Looks $. This is gonna really help us track down issues faster.  I really like your `pathToContext` idea too. Merging..

File: lib/utils/aws.js [hunk 1]
```diff
@@ -84,15 +84,13 @@ module.exports.profilesMap = function() {
...
+      configPath = path.join(configDir, 'config');
 
-  if (!utils.fileExistsSync(credsPath)) {
-    utils.writeFile(credsPath);
-  }
-
-  if (!utils.fileExistsSync(configPath)) {
-    utils.writeFile(configPath);
+  if (!utils.dirExistsSync(configDir)) {
+    fs.mkdirSync(configDir, 0o700);
```
Comment (doapp-ryanp) [inline lib/utils/aws.js]: Just a heads up, we are not forcing es6 yet. had to change this to es5 compat.

Comment (boushley) [inline lib/utils/aws.js]: Whoops sorry about that. Using node 4.1 locally so I didn't even realize.

Thanks for fixing, just the number format?

Aaron

**Pull request merged**